File: Selenium/Activities/Activity9.py
# Import required packages from selenium
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/ajax")
    home_title=driver.title
    print ("Title of the Home Page is : {}".format(home_title))

    # Get the change Button and Click on it
    changeButton = driver.find_element(By.CSS_SELECTOR, "button.ui.violet.button")
    changeButton.click()

    dynamic_text = driver.find_element(By.TAG_NAME, "h1")

    # Wait till the Hello! appears and print it
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME, "h1"), "HELLO!"))
    print("Dynamic Text : " + dynamic_text.text)

    # Wait till the I'm Late! appears and print it
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME, "h3"), "I'm late!"))

    lateText = driver.find_element(By.TAG_NAME, "h3")
    print("Dynamic Text : " + lateText.text)

    # Wait for 5 Seconds and Close the Browser
    time.sleep(3)
    driver.close()

File: Selenium/Activities/Activity9.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity9 {
    public static void main(String[] args) throws InterruptedException {
        //Setup GekoDriver
        WebDriverManager.firefoxdriver().setup();

        //Create driver Object for Firefox Browser
        WebDriver driver = new FirefoxDriver();

        //Set Explicit wait of 5 Secs
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofMillis(5000));

        //Open the url
        driver.get("https://v1.training-support.net/selenium/ajax");

        //Get the Title
        String title = driver.getTitle();
        System.out.print("Title of the Page is : " + title + "\n");

        // Get the change Button and Click on it
        WebElement changeButton = driver.findElement(By.cssSelector("button.ui.violet.button"));
        changeButton.click();

        WebElement dynamicText = driver.findElement(By.tagName("h1"));

        // Wait till the Hello! appears and print it
        wait.until(ExpectedConditions.textToBePresentInElement(dynamicText, "HELLO!"));
        System.out.println("Dynamic Text : " + dynamicText.getText());

        // Wait till the I'm Late! appears and print it
        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.tagName("h3"), "I'm late!"));

        WebElement lateText = driver.findElement(By.tagName("h3"));
        System.out.println("Dynamic Text : " + lateText.getText());

        //Wait for 5 Seconds and Close the Browser
        Thread.sleep(5000);
        driver.close();
    }
}
